apps/mobile/src/libs/components/gradient-slider/gradient-slider.tsx

```diff
@@ -0,0 +1,132 @@
+} from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import {
+	generateSliderGradientColors,
+	generateSliderGradientLocations,
+} from "~/libs/helpers/helpers";
+import { useCallback, useState } from "~/libs/hooks/hooks";
+import {
+	type colorToGradientColors,
+	directionToGradient,
+} from "~/libs/maps/maps";
+import { globalStyles } from "~/libs/styles/styles";
+import { type LayoutChangeEvent } from "~/libs/types/types";
+
+import { styles } from "./styles";
+
+type Properties = {
+	gradientColors: keyof typeof colorToGradientColors;
+	max: number;
+	min: number;
+	onValueChange?: (value: number) => void;
+};
+
+const INITIAL_SLIDER_WIDTH = 0;
+const MAX_PERCENT = 100;
+const PLANET_TRANSLATE_OFFSET = 3;
+const ICON_TRANSLATE_OFFSET = 20;
+const MARKER_ICON_SIZE = 50;
+const SLIDER_STEP = 1;
+
+const GradientSlider: React.FC<Properties> = ({
+	gradientColors,
+	max,
+	min,
+	onValueChange,
+}) => {
+	const [value, setValue] = useState<number>(min);
+	const [sliderWidth, setSliderWidth] = useState<number>(INITIAL_SLIDER_WIDTH);
+	const initialColors = generateSliderGradientColors(min, gradientColors, max);
+	const [color, setColor] = useState<string[]>(initialColors);
+
+	const handleLayout = useCallback((event: LayoutChangeEvent): void => {
+		const { width } = event.nativeEvent.layout;
+		setSliderWidth(width);
+	}, []);
+
+	const handleValueChange = useCallback(
+		(sliderValue: number) => {
+			setValue(sliderValue);
+			setColor(generateSliderGradientColors(sliderValue, gradientColors, max));
+
+			if (onValueChange) {
+				onValueChange(sliderValue);
+			}
+		},
+		[gradientColors, max, onValueChange],
+	);
+
+	const pixelsPerPercent = sliderWidth / MAX_PERCENT;
+	const translateXBase = (value / max) * MAX_PERCENT * pixelsPerPercent;
+
+	const translateXForPlanet = translateXBase - PLANET_TRANSLATE_OFFSET;
+	const translateXForIcon = translateXBase - ICON_TRANSLATE_OFFSET;
+
+	const { end, start } = directionToGradient.leftToRight;
+
+	return (
+		<View style={[globalStyles.alignItemsCenter, globalStyles.flex1]}>
+			<View
+				onLayout={handleLayout}
+				style={[globalStyles.justifyContentCenter, styles.sliderContainer]}
+			>
+				<LinearGradient
+					colors={color}
+					end={end}
+					locations={generateSliderGradientLocations(max)}
+					start={start}
+					style={styles.gradient}
+				/>
+				<View
+					style={[
+						globalStyles.alignItemsCenter,
+						globalStyles.justifyContentStart,
+						styles.labelContainer,
+						{
+							transform: [{ translateX: translateXForIcon }],
+						},
+					]}
+				>
+					<Icon color={BaseColor.BLACK} name="place" size={MARKER_ICON_SIZE} />
+					<Text preset="regular" style={[globalStyles.mt12, styles.labelText]}>
+						{value}
+					</Text>
+				</View>
+
+				<Planet
+					color={gradientColors}
+					size="slider"
+					style={[
+						styles.planet,
+						{
+							transform: [{ translateX: translateXForPlanet }],
+						},
+					]}
+				/>
```

Comment:
If it is used for slider sections, then it's not a good approach. It's better to create `SliderSection` component for this and leave the Planet as it is.

Reply:
I am using the planet as a slider indicator for the track — instead of a solid color, I'm using our custom gradient. I’m not entirely sure what exactly should be moved

Reply:
I think @revenets meant that Planet is currently used in the context of a slider with the size="slider" parameter.
It would be better to create a separate component (SliderSection) that will be responsible for everything related to the slider. That is, you can take this part out separately:

```
<View style={styles.container}>
      <Planet
        color={gradientColors}
        size= // some default size for slider
        style={[
          styles.planet, 
          { transform: [{ translateX: translateXForPlanet }] }
        ]}
      />
     // maybe other elements of slider
</View>
```

Reply:
That's not exactly what I meant. The main point was that the Planet component was styled and created with this exact purpose in mind. And modifying it to fit a completely different approach even if it uses similar styles as you need is not correct. That's why I suggest you to create a separate component `SliderSection ` with specific props and styles you need for Slider.